### PR TITLE
Add geolocation fields to the config

### PR DESCRIPTION
**Pull request by andrenerd**
@Sriep 
Need you input. Thanks.

- how about to move the GeolocationConfig struct to 0chain main app, specifically here (and import from there to specific services, such as blobber):
https://github.com/0chain/0chain/blob/master/code/go/0chain.net/chaincore/config/config.go

- it was also mentioned to add geolocation props to storagsc and minersc smart contracts too. does it mean, the props should be added to these structs specifically (currently they have near to zero fields)?
https://github.com/0chain/0chain/blob/ee5704d2617f42dcf857270cf73be44a9e5c523b/code/go/0chain.net/smartcontract/storagesc/sc.go#L24
https://github.com/0chain/0chain/blob/ee5704d2617f42dcf857270cf73be44a9e5c523b/code/go/0chain.net/smartcontract/minersc/sc.go#L51

### CODE REVIEW

**Comment by Sriep** [comment]
Needs to add geo-location to blobber.yaml, and pass the geo-location information to 0cahin when registering blobber.

Part of https://github.com/0chain/0chain/issues/238. There is no urgency to merge this Pr until the `0cahin` and `zboxcli` parts of issue 238 are ready.

**Comment by Sriep** [comment]
I think we should [validate longitude and latitude values.](https://stackoverflow.com/questions/15965166/what-are-the-lengths-of-location-coordinates-latitude-and-longitude). Confirm:
* Latitude : max/min 90.0000000 to -90.0000000
* Longitude : max/min 180.0000000 to -180.0000000
Should be good enough.

**Comment by andrenerd** [comment]
@Sriep thanks. added. please, recheck.
ps. not found better place to put the validation, any suggestion are welcome.

**Review by Sriep — CHANGES_REQUESTED** [review]

**Comment by Sriep** [comment]
Merge in the latest master changes, as I think that might fix the failing unit test check.

File: code/go/0chain.net/blobbercore/config/config.go [hunk 1]
```diff
@@ -133,6 +141,17 @@ func Development() bool {
 	return Configuration.DeploymentMode == DeploymentDevelopment
 }
 
+// get validated geolocatiion
+func Geolocation() GeolocationConfig {
+	g := Configuration.Geolocation
+	if g.Latitude > 90.00 || g.Latitude < -90.00 ||
+		g.Longitude > 180.00 || g.Longitude < -180.00 {
+			g.Latitude = float64(0.00)
```
[the remark this answers is not in the document]
Comment (andrenerd) [reply code/go/0chain.net/blobbercore/config/config.go:149]: Ok. If there should be error, let it be panic error.
As passing regular error and forcing higher level components to deal with it makes no sense.

File: code/go/0chain.net/core/transaction/entity.go [hunk 2]
```diff
@@ -68,13 +68,21 @@ type StakePoolSettings struct {
 	ServiceCharge float64 `json:"service_charge"`
 }
 
+// Move to the core one day
+type StorageNodeGeolocation struct {
+	Latitude float64 `json:"latitude"`
+	Longitude float64 `json:"longitude"`
+	// reserved / Accuracy float64 `mapstructure:"accuracy"`
```
[the remark this answers is not in the document]
Comment (andrenerd) [reply code/go/0chain.net/core/transaction/entity.go:75]: Everything except this point will be update. I prefer to keep this, as a clue for potential upgrades.

**Comment by Sriep** [comment]
Please merge in the latest master changes. You need the fix to the unit test checks.

File: code/go/0chain.net/blobbercore/config/config.go [hunk 3]
```diff
@@ -63,6 +63,12 @@ const (
 	DeploymentMainNet     = 2
 )
 
+type GeolocationConfig struct {
+	Latitude float64 `mapstructure:"latitude"`
+	Longitude float64 `mapstructure:"longitude"`
+	// reserved / Accuracy float64 `mapstructure:"accuracy"`
```
Comment (Sriep) [inline code/go/0chain.net/blobbercore/config/config.go:69]: Please remove. It just confuses people and uses up space.

File: code/go/0chain.net/blobbercore/readmarker/entity.go [hunk 4]
```diff
@@ -106,7 +106,6 @@ func GetLatestReadMarkerEntity(ctx context.Context, clientID string) (*ReadMarke
 }
 
 func SaveLatestReadMarker(ctx context.Context, rm *ReadMarker, isCreate bool) error {
-
 	var (
```
Comment (Sriep) [inline code/go/0chain.net/blobbercore/readmarker/entity.go:109]: If there are no meaningful changes to entity.go, revert changes and remove from PR.

File: code/go/0chain.net/core/transaction/entity.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (Sriep) [reply code/go/0chain.net/core/transaction/entity.go:75]: Remove unnecessary comments.It just confuses people.

**Review by Sriep — CHANGES_REQUESTED** [review]

**Pull request merged**